python/hail/methods/statgen.py

```diff
@@ -1454,16 +1452,10 @@ def pca(entry_expr, k=10, compute_loadings=False, as_array=False) -> Tuple[List[
     (:obj:`list` of :obj:`float`, :class:`.Table`, :class:`.Table`)
         List of eigenvalues, table with column scores, table with row loadings.
     """
-    source = entry_expr._indices.source
-    if not isinstance(source, MatrixTable):
-        raise ValueError("Expect an expression of 'MatrixTable', found {}".format(
-            "expression of '{}'".format(source.__class__) if source is not None else 'scalar expression'))
-    dataset = source
-    base, _ = dataset._process_joins(entry_expr)
-    analyze('pca', entry_expr, dataset._entry_indices)
-
-    r = Env.hail().methods.PCA.apply(dataset._jvds, to_expr(entry_expr)._ast.to_hql(), k, compute_loadings, as_array)
-    scores = Table(Env.hail().methods.PCA.scoresTable(dataset._jvds, as_array, r._2()))
+    mt = analyze_entry_expr('pca/entry_expr', entry_expr)
+
+    r = Env.hail().methods.PCA.apply(mt._jvds, to_expr(entry_expr)._ast.to_hql(), k, compute_loadings, as_array)
```

Comment:
process joins is definitely required here. The solution would be to rework PCA to take a field name.

Reply:
you didn't address this, I think

Reply:
and can you add a test of something that uses a `literal`?

like `annotate-entries(x = x * hl.literal([1])[0])`